### ISSUE TITLE
dartfix --help throws exception when run with dev & stable SDKs

### ISSUE THREAD

**Issue by jcollins-g** [dart-lang/sdk#41679]
Example output from stable:

```
jcollins-macbookpro2:dartdoc jcollins$ pub global activate dartfix
Package dartfix is currently active at version 0.1.6.
Resolving dependencies... 
+ analysis_server_client 1.1.2
+ args 1.6.0
+ cli_util 0.1.3+2
+ collection 1.14.12
+ dartfix 0.1.6
+ path 1.7.0
+ pub_semver 1.4.4
Precompiling executables... (1.5s)
Precompiled dartfix:dartfix.
Installed executable dartfix.
Activated dartfix 0.1.6.
jcollins-macbookpro2:dartdoc jcollins$ dartfix --help
Usage: dartfix [options...] <directory paths>

Choosing fixes to be applied:
    --fix=<name-of-fix>           Include a specific fix.
    --excludeFix=<name-of-fix>    Exclude a specific fix.
    --pedantic                    Apply pedantic fixes.
    --required                    Apply required fixes.

Modifying files:
-w, --overwrite                   Overwrite files with the changes.
-f, --force                       Overwrite files even if there are errors.

Miscellaneous:
-h, --help                        Display this help message.
-v, --verbose                     Verbose output.
    --[no-]color                  Use ansi colors when printing messages.
                                  (defaults to on)

Listing fixes...               0.4s

The following fixes are automatically applied unless at least one --fix option is specified
(and --required is not specified). They may be individually disabled using --excludeFix.
Unhandled exception:
Failed assertion: boolean expression must not be null
#0      WhereIterator.moveNext (dart:_internal/iterable.dart)
#1      Iterable.forEach (dart:core/iterable.dart:279:23)
#2      Driver.showFixes (package:dartfix/src/driver.dart:221:42)
<asynchronous suspension>
#3      Driver.start (package:dartfix/src/driver.dart:268:15)
<asynchronous suspension>
#4      main (file:///Users/jcollins/.pub-cache/hosted/pub.dartlang.org/dartfix-0.1.6/bin/dartfix.dart:13:17)
#5      _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#6      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
jcollins-macbookpro2:dartdoc jcollins$ dart --version
Dart VM version: 2.8.0 (stable) (Thu Apr 23 13:05:36 2020 +0200) on "macos_x64"
jcollins-macbookpro2:dartdoc jcollins$
```

**Comment by jcollins-g** [comment]
It looks like this works with head.  We will need to publish dartfix and possibly analysis_server_client.

**Comment by stereotype441** [comment]
https://dart-review.googlesource.com/c/sdk/+/148020 removes the dependency from dartfix to nnbd_migration, so I believe that once that is landed, it should be possible to publish dartfix.

**Comment by stereotype441** [comment]
Ok, https://dart-review.googlesource.com/c/sdk/+/148020 was landed as d00d800f3a75bafcbb0e0e9cee49b20067016e68, so I believe it should now be possible to publish dartfix again.  Reassigning back to Janice.

**Comment by jcollins-g** [comment]
dartfix 0.1.7 has been published, verified the issue is fixed.
